67_RayQueryGeometry/app_resources/render.comp.hlsl

```diff
@@ -0,0 +1,160 @@
+#include "common.hlsl"
+
+#include "nbl/builtin/hlsl/spirv_intrinsics/core.hlsl"
+
+struct SPushConstants
+{
+    SGeomBDA geom[OT_COUNT];
+};
+
+[[vk::push_constant]] SPushConstants pc;
+
+[[vk::binding(0, 0)]]
+RaytracingAccelerationStructure topLevelAS;
+
+[[vk::binding(1, 0)]]
+cbuffer CameraData
+{
+    SCameraParameters params;
+};
+
+[[vk::binding(2, 0)]] RWTexture2D<float4> outImage;
+
+float3 unpackNormals3x10(uint32_t v)
+{
+    // host side changes float32_t3 to EF_A2B10G10R10_SNORM_PACK32
+    float3 n = (float3)0;
+    const uint mask = 0x3FF;    // 10 bits
+    n.x = float(asint((v >> 20) & mask));
+    n.y = float(asint((v >> 10) & mask));
+    n.z = float(asint((v >> 0) & mask));
+    return n;
```

Comment:
normals are always snorm and not unorm, snorm unpacks differently to unrom (its not a simple *0.5-0.5 from a unorm)